File: prob22.R
lst <- scan("names.txt", what = "", sep = ",")
lst[is.na(lst)] <- "NA"  # "NA" as a name? REALLY?!
lst <- sort(lst)
lst.len <- length(lst)
 
let2num <- data.frame(row.names = LETTERS, NUM = 1:26)  # the converting table
 
NameScore <- function(name) {
  # helper function to convert letter to its alphabetical order and sum up
  name.char <- unlist(strsplit(name, split = ""))
  score <- sum(let2num[name.char, "NUM"])
  return(score)
}
 
name.scores <- numeric(lst.len)
for (i in 1:lst.len) {
  name.scores[i] <- NameScore(lst[i])
}
result <- sum(name.scores * (1:lst.len))
cat("The result is:", result, "\n")

File: prob22.py
import string
 
namesFile = open("names.txt","r")
names = namesFile.read()
names = names.replace('"', "") 
names = names.split(",")
names.sort()
 
 
 
totalScore = 0
for name in names:
    nameValue = 0
    for letter in name:
        nameValue += string.ascii_uppercase.index(letter) + 1
 
    totalScore += (names.index(name) + 1) * nameValue
 
print totalScore
